tests/AutoReview/ProjectCode/ProjectCodeProvider.php

```diff
@@ -0,0 +1,292 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\AutoReview\ProjectCode;
+
+use Generator;
+use Infection\Command\ConfigureCommand;
+use Infection\Command\InfectionCommand;
+use Infection\Config\ConsoleHelper;
+use Infection\Config\Guesser\SourceDirGuesser;
+use Infection\Config\InfectionConfig;
+use Infection\Console\Application;
+use Infection\Console\OutputFormatter\OutputFormatter;
+use Infection\Console\OutputFormatter\ProgressFormatter;
+use Infection\Console\Util\PhpProcess;
+use Infection\Differ\DiffColorizer;
+use Infection\Differ\Differ;
+use Infection\Finder\ComposerExecutableFinder;
+use Infection\Finder\TestFrameworkFinder;
+use Infection\Http\BadgeApiClient;
+use Infection\Logger\ResultsLoggerTypes;
+use Infection\Mutant\MetricsCalculator;
+use Infection\Mutator\Util\Mutator;
+use Infection\Process\Builder\ProcessBuilder;
+use Infection\Process\Listener\MutantCreatingConsoleLoggerSubscriber;
+use Infection\Process\Listener\MutationGeneratingConsoleLoggerSubscriber;
+use Infection\Process\Runner\MutationTestingRunner;
+use Infection\TestFramework\Coverage\CodeCoverageData;
+use Infection\TestFramework\Coverage\CoverageFileData;
+use Infection\TestFramework\Coverage\CoverageLineData;
+use Infection\TestFramework\Coverage\CoverageMethodData;
+use Infection\TestFramework\Coverage\TestFileTimeData;
+use Infection\TestFramework\PhpSpec\Config\Builder\InitialConfigBuilder as PhpSpecInitalConfigBuilder;
+use Infection\TestFramework\PhpSpec\Config\Builder\MutationConfigBuilder as PhpSpecMutationConfigBuilder;
+use Infection\TestFramework\PhpSpec\Config\NoCodeCoverageException;
+use Infection\TestFramework\PhpUnit\Config\Builder\InitialConfigBuilder as PhpUnitInitalConfigBuilder;
+use Infection\TestFramework\PhpUnit\Config\Builder\MutationConfigBuilder as PhpUnitMutationConfigBuilder;
+use Infection\TestFramework\PhpUnit\Coverage\CoverageXmlParser;
+use Infection\TestFramework\TestFrameworkTypes;
+use function Infection\Tests\generator_to_phpunit_data_provider;
+use Infection\Utils\VersionParser;
+use Infection\Visitor\MutationsCollectorVisitor;
+use Infection\Visitor\ParentConnectorVisitor;
+use function iterator_to_array;
+use ReflectionClass;
+use const SORT_STRING;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+final class ProjectCodeProvider
+{
+    /**
+     * List of classes which should not have a matching test.
```

Comment:
Looking at the old class i notice that this never had a propper comment.

These classes are not *intended* to not have tests. Its simply that this is somewhat 'legacy' and they aren't tested *yet*. And i think this comment should reflect that.

Reply:
I'm open to suggestion on a wording, I would not however block the PR on that.

Reply:
If we somewhat mimic the `NON_FINAL_EXTENSION_CLASSES` a bit here we can do the following:
```
   /**
     * This array contains all classes that don't have tests yet, due to legacy
     * reasons. This list should never be added to, only removed from.
     */
```